sdks/java/core/src/main/java/org/apache/beam/sdk/util/common/Counter.java

```diff
@@ -295,6 +296,56 @@ public AggregationKind getKind() {
   public abstract CounterMean<T> getMean();
 
   /**
+   * CommitState represents whether counters' values have been committed to the backend.
+   *
+   * <p>Runners can use this information to optimize counters updates.
+   * For example, if counters are committed, runners may choose to skip the updates.
+   *
+   * <p>Counters' state transition table:
```

Comment:
If Counter is not part of the model, then we should be able to move it into the runner -- the runner can depend on Aggregator. DirectPipelineRunner may need a different implementation, but that should be fine since it can just implement Aggregator directly.